src/rapids_pre_commit_hooks/pyproject_license.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import random
+
+import tomlkit
+import tomlkit.exceptions
+
+from .lint import LintMain
+
+RAPIDS_LICENSE = "Apache 2.0"
+
+
+def find_value_location(document, key):
+    copied_document = copy.deepcopy(document)
+    node = copied_document
+    while str(placeholder := random.randint(0, 1048576)) in node.as_string():
+        pass
```

Comment:
So you're just looping until you find a random number that isn't anywhere in the document? In principle this could fail forever with a specifically constructed file, but I guess the edge case is so remote that it's not worth guarding against an infinite loop here.

Reply:
To collide with every single number between `0` and `1048576`, the `pyproject.toml` document would have to be a minimum of about a megabyte of nothing but numbers. While it's theoretically possible, it's not something I would ever expect to encounter in a RAPIDS project.

Reply:
There probably should be some guard to break the loop. It's certainly unlikely to be a problem in regular usage, but a malicious user could push a change and cause CI to churn a lot. Though I think there is an absolute time limit on jobs.

Reply:
Good point about malicious users. I'll add some kind of limit.